README.md

```diff
@@ -188,38 +213,110 @@ We've recovered from the conversion error and kept going. Algebraic effects and
+
+  val match_with: ('c -> 'a) -> 'c -> ('a,'b) handler -> 'b
+  (** [match_with f v h] runs the computation [f v] in the handler [h]. *)
+
+  type 'a effect_handler =
+    { effc: 'b. 'b t -> (('b, 'a) continuation -> 'a) option }
+  (** ['a effect_handler] is a deep handler with an identity value handler
+      [fun x -> x] and an exception handler that raises any exception
+      [fun e -> raise e]. *)
+
+  val try_with: ('b -> 'a) -> 'b -> 'a effect_handler -> 'a
+  (** [try_with f v h] runs the computation [f v] under the handler [h]. *)
+end
+
+module Shallow : sig
+  (** Some contents omitted *)
+
+  type ('a,'b) handler =
+    { retc: 'a -> 'b;
+      exnc: exn -> 'b;
+      effc: 'c.'c t -> (('c,'a) continuation -> 'b) option }
+  (** [('a,'b) handler] is a handler record with three fields -- [retc]
+      is the value handler, [exnc] handles exceptions, and [effc] handles the
+      effects performed by the computation enclosed by the handler. *)
+
+  val continue_with : ('c,'a) continuation -> 'c -> ('a,'b) handler -> 'b
+  (** [continue_with k v h] resumes the continuation [k] with value [v] with
+      the handler [h].
+      @raise Continuation_already_resumed if the continuation has already been
+      resumed.
+   *)
+end
+```
+
+The handlers are records with three fields and are called in the context of `match_with`, `try_with`, or `continue_with`:
+
+`retc` is the function that is called when the computation returns a value -
+i.e. no effects or exceptions were performed/raised in the computation. The
+function has one parameter: the value of the computation
+
+`exnc` is called when the computation throws an exception. It takes the exception as a parameter.
+
+`effc` is the function that handles the effects. It has type `'c. 'c Effect.t -> ('c, 'a) continuation -> 'b) option`
+
+Effects are strongly typed, but the handler function can handle multiple
+effects and has to be generic over every possible type (which is potentially
+all of them since the effects variant can always be extended further), hence
+the `'c` existential type. `effc` returns an `option` where a None value means
+ignore the effect (and crash the program if not handled somewhere else). A Some
+value holds a function that takes a parameter commonly called `k`
 
 ```ocaml
-try sum_up r with
-| End_of_file -> Printf.printf "Sum is %d\n" !r
-| effect (Conversion_failure s) k ->
-    Printf.fprintf stderr "Conversion failure \"%s\"\n%!" s;
-    continue k 0
+  { effc = (fun (type c) (eff: c Effect.t) ->
+      match eff with
+      | Conversion_failure s -> Some (fun (k: (c,_) continuation) ->
+              Printf.fprintf stderr "Conversion failure \"%s\"\n%!" s;
+              continue k 0)
+      | _ -> None
+    )
+   }
 ```
 
-but has an additional parameter `k`, which is the *delimited continuation*
+We need to declare a [locally abstract type](https://kcsrk.info/webman/manual/locallyabstract.html) `c` in order to
```

Comment:
I suggest we link to the official copy of the manual instead, i.e. https://v2.ocaml.org/manual/locallyabstract.html

Reply:
Done